### ISSUE TITLE
Boost fails to build

### ISSUE THREAD

**Issue by JBaczuk** [JBaczuk/projenpp#18]
```
cpp_miner_tests-runner.o: In function `_GLOBAL__sub_I__Z14init_unit_testv':
/usr/include/boost/test/unit_test_log.hpp:217: undefined reference to `vtable for boost::unit_test::unit_test_log_t'
cpp_miner_tests-runner.o: In function `main':
/usr/include/boost/test/unit_test.hpp:63: undefined reference to `boost::unit_test::unit_test_main(bool (*)(), int, char**)'
cpp_miner_tests-cpp_miner_tests.o: In function `test_util::test_isSpace::test_method()':
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:9: undefined reference to `isSpace(char)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:10: undefined reference to `boost::unit_test::unit_test_log_t::set_checkpoint(boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::basic_cstring<char const>)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:10: undefined reference to `boost::test_tools::tt_detail::report_assertion(boost::test_tools::assertion_result const&, boost::unit_test::lazy_ostream const&, boost::unit_test::basic_cstring<char const>, unsigned long, boost::test_tools::tt_detail::tool_level, boost::test_tools::tt_detail::check_type, unsigned long, ...)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:11: undefined reference to `isSpace(char)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:12: undefined reference to `boost::unit_test::unit_test_log_t::set_checkpoint(boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::basic_cstring<char const>)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:12: undefined reference to `boost::test_tools::tt_detail::report_assertion(boost::test_tools::assertion_result const&, boost::unit_test::lazy_ostream const&, boost::unit_test::basic_cstring<char const>, unsigned long, boost::test_tools::tt_detail::tool_level, boost::test_tools::tt_detail::check_type, unsigned long, ...)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:13: undefined reference to `isSpace(char)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:14: undefined reference to `boost::unit_test::unit_test_log_t::set_checkpoint(boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::basic_cstring<char const>)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:14: undefined reference to `boost::test_tools::tt_detail::report_assertion(boost::test_tools::assertion_result const&, boost::unit_test::lazy_ostream const&, boost::unit_test::basic_cstring<char const>, unsigned long, boost::test_tools::tt_detail::tool_level, boost::test_tools::tt_detail::check_type, unsigned long, ...)'
cpp_miner_tests-cpp_miner_tests.o: In function `test_util::test_isSpace_invoker()':
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:7: undefined reference to `boost::unit_test::unit_test_log_t::set_checkpoint(boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::basic_cstring<char const>)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:7: undefined reference to `boost::unit_test::unit_test_log_t::set_checkpoint(boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::basic_cstring<char const>)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:7: undefined reference to `boost::unit_test::unit_test_log_t::set_checkpoint(boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::basic_cstring<char const>)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:7: undefined reference to `boost::unit_test::unit_test_log_t::set_checkpoint(boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::basic_cstring<char const>)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:7: undefined reference to `boost::unit_test::unit_test_log_t::set_checkpoint(boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::basic_cstring<char const>)'
cpp_miner_tests-cpp_miner_tests.o: In function `__static_initialization_and_destruction_0':
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:5: undefined reference to `boost::unit_test::ut_detail::auto_test_unit_registrar::auto_test_unit_registrar(boost::unit_test::basic_cstring<char const>, boost::unit_test::basic_cstring<char const>, unsigned long, boost::unit_test::decorator::collector&)'
cpp_miner_tests-cpp_miner_tests.o: In function `_GLOBAL__sub_I__ZN9test_util12test_isSpace11test_methodEv':
/usr/include/boost/test/tree/test_unit.hpp:249: undefined reference to `boost::unit_test::ut_detail::normalize_test_case_name[abi:cxx11](boost::unit_test::basic_cstring<char const>)'
/usr/include/boost/test/tree/test_unit.hpp:249: undefined reference to `boost::unit_test::test_case::test_case(boost::unit_test::basic_cstring<char const>, boost::unit_test::basic_cstring<char const>, unsigned long, boost::function<void ()> const&)'
cpp_miner_tests-cpp_miner_tests.o: In function `__static_initialization_and_destruction_0':
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:7: undefined reference to `boost::unit_test::ut_detail::auto_test_unit_registrar::auto_test_unit_registrar(boost::unit_test::test_case*, boost::unit_test::decorator::collector&, unsigned long)'
/home/jbaczuk/cpp_miner/test/cpp_miner_tests.cpp:17: undefined reference to `boost::unit_test::ut_detail::auto_test_unit_registrar::auto_test_unit_registrar(int)'
cpp_miner_tests-cpp_miner_tests.o: In function `_GLOBAL__sub_I__ZN9test_util12test_isSpace11test_methodEv':
/usr/include/boost/test/unit_test_log.hpp:217: undefined reference to `vtable for boost::unit_test::unit_test_log_t'
collect2: error: ld returned 1 exit status
Makefile:300: recipe for target 'cpp_miner_tests' failed
make[2]: *** [cpp_miner_tests] Error 1
make[2]: Leaving directory '/home/jbaczuk/cpp_miner/test'
Makefile:437: recipe for target 'check-am' failed
make[1]: *** [check-am] Error 2
make[1]: Leaving directory '/home/jbaczuk/cpp_miner/test'
Makefile:562: recipe for target 'check-recursive' failed
```

This should fail in configure if it is not installed properly.

**Comment by JBaczuk** [comment]
Will fix configure script in #7

**Comment by JBaczuk** [comment]
Fixed in #12
